src/commands/game_commands/preset.ts

```diff
@@ -142,22 +191,88 @@ export default class PresetCommand implements BaseCommand {
     }
 
     async replacePreset(presetName: string, guildPreference: GuildPreference, messageContext: MessageContext) {
-        const deleteResult = await guildPreference.deletePreset(presetName);
-        if (!deleteResult) {
+        const oldUUID = await guildPreference.deletePreset(presetName);
+        if (!oldUUID) {
             logger.info(`${getDebugLogHeader(messageContext)} | Preset '${presetName}' replace, preset did not exist`);
         }
 
-        await guildPreference.savePreset(presetName);
+        await guildPreference.savePreset(presetName, oldUUID);
         logger.info(`${getDebugLogHeader(messageContext)} | Preset '${presetName}' successfully replaced`);
         await sendInfoMessage(messageContext, { title: "Preset Replaced", description: `You can load this preset later with \`${process.env.BOT_PREFIX}preset load ${presetName}\`.`, thumbnailUrl: KmqImages.HAPPY });
     }
 
+    async exportPreset(presetName: string, guildPreference: GuildPreference, messageContext: MessageContext) {
+        const presetUUID = await guildPreference.getPresetUUID(presetName);
+        if (!presetUUID) {
+            await sendInfoMessage(messageContext, { title: "Preset Export Failed", description: `The given preset \`${presetName}\` does not exist.`, thumbnailUrl: KmqImages.DEAD });
+            logger.warn(`${getDebugLogHeader(messageContext)} | Preset export failed; '${presetName}' does not exist`);
+            return;
+        }
+
+        await sendInfoMessage(messageContext, { title: "Preset Exported Successfully", description: `Import \`${presetName}\` as a new preset in other servers using:\n\`${process.env.BOT_PREFIX}preset import ${presetUUID} [preset_name]\`\n\nAlternatively, load the preset's options directly using:\n\`${process.env.BOT_PREFIX}preset load ${presetUUID}\``, thumbnailUrl: KmqImages.THUMBS_UP });
+        logger.info(`${getDebugLogHeader(messageContext)} | Preset '${presetName}' successfully exported as ${presetUUID}`);
+    }
+
+    async importPreset(presetUUID: string, presetName: string, guildPreference: GuildPreference, messageContext: MessageContext) {
+        if ((await guildPreference.listPresets()).includes(presetName)) {
+            sendErrorMessage(messageContext, { title: "Preset Import Failed", description: `Preset \`${presetName}\` already exists. You can delete the old one with \`${process.env.BOT_PREFIX}preset delete ${presetName}\`.`, thumbnailUrl: KmqImages.DEAD });
+            logger.warn(`${getDebugLogHeader(messageContext)} | Preset import failed; '${presetName}' already exists`);
+            return;
+        }
+
+        const existingPresetIdentifiers = await dbContext.kmq("game_option_presets")
```

Comment:
is this supposed to be plural

Reply:
I decided to call it that since the two identifiers are guild_id and preset_name